src/hooks/useGetProfileImg.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react';
+import { useRecoilState } from 'recoil';
+import { getDoc } from 'firebase/firestore';
+import { getUserDocRef } from '@/api/profile';
+import { imageState } from '@/atoms/atoms';
+import useGetCacheData from '@/hooks/useGetCacheData';
+
+export const useGetProfileImg = () => {
+  const [profileUrl, setProfileUrl] = useRecoilState(imageState);
+  const userId = useGetCacheData('user', '/userId');
+
+  useEffect(() => {
+    const fetchProfileImg = async () => {
+      if (!userId) {
+        return;
+      }
+      const userDocRef = await getUserDocRef();
+      try {
+        const fetchedImg = await getDoc(userDocRef);
+        if (fetchedImg.exists()) {
+          const userData = fetchedImg.data();
+          const userProfileUrl = userData?.user?.profileUrl || '';
+          setProfileUrl(userProfileUrl);
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    fetchProfileImg();
+  }, [setProfileUrl]);
+};
```

Comment:
얼리리턴부터 예외처리까지 깔끔하게 하셨네용 다만 `try-catch` 를 사용한 김에 indent depth가 늘리는 20번 라인보다는 throw를 활용한다면 어떨까요? 얼리리턴과 동일한 효과지만 return이 아닌 catch로 넘기면서 예외처리의 목적에 맞는 코드가 될 것 같습니다.
```suggestion
import { useEffect } from 'react';
import { useRecoilState } from 'recoil';
import { getDoc } from 'firebase/firestore';
import { getUserDocRef } from '@/api/profile';
import { imageState } from '@/atoms/atoms';
import useGetCacheData from '@/hooks/useGetCacheData';

export const useGetProfileImg = () => {
  const [profileUrl, setProfileUrl] = useRecoilState(imageState);
  const userId = useGetCacheData('user', '/userId');

  useEffect(() => {
    const fetchProfileImg = async () => {
      if (!userId) {
        return;
      }
      const userDocRef = await getUserDocRef();
      try {
        const fetchedImg = await getDoc(userDocRef);
        if (!fetchedImg.exists()) throw new Error('이미지가 존재하지 않습니다.');
        const userData = fetchedImg.data();
        const userProfileUrl = userData?.user?.profileUrl || '';
        setProfileUrl(userProfileUrl);
      } catch (err) {
        console.log(err);
      }
    };
    fetchProfileImg();
  }, [setProfileUrl]);
};
```

Reply:
오호 코드를 작성할 때 시야가 좁아져 해당 부분은 미처 생각하지 못했던 것 같네요 !
유사 작업이 필요했는데, 해당 부분 참고하여 적용해보겠습니다 😎